packages/api/src/clickhouse/__tests__/clickhouse.test.ts

```diff
@@ -751,6 +751,181 @@ Array [
     "ts_bucket": 1641341100,
   },
 ]
+`);
+
+    const buildMetricSeriesHistogram = ({
+      tags,
+      name,
+      points,
+      data_type,
+      is_delta,
+      is_monotonic,
+      unit,
+    }: {
+      tags: Record<string, string>;
+      name: string;
+      points: { value: number; timestamp: string | number; le: string }[];
+      data_type: clickhouse.MetricsDataType;
+      is_monotonic: boolean;
+      is_delta: boolean;
+      unit: string;
+    }): MetricModel[] => {
+      return points.map(({ value, timestamp, le }) => ({
+        _string_attributes: { ...tags, le: le.toString() },
+        name,
+        value,
+        timestamp: parseInt(timestamp.toString(), 10) * 1000000,
+        data_type,
+        is_monotonic,
+        is_delta,
+        unit,
+      }));
+    };
+
+    await clickhouse.bulkInsertTeamMetricStream(
+      buildMetricSeriesHistogram({
+        name: 'test.response_time',
+        tags: { host: 'test2', runId },
+        data_type: clickhouse.MetricsDataType.Histogram,
+        is_monotonic: false,
+        is_delta: false,
+        unit: '',
+
+        // OK this is a fun one. We want a histogram with a bunch of buckets
+        // but we need to make sure that the quantile value is some reasonable value
+        // So here's a distribution where the p50 is 50ms, p90 is 100ms, p99 is 200ms
+        points: [
+          { value: 10, timestamp: now, le: '10' },
+          { value: 20, timestamp: now, le: '30' },
+          { value: 25, timestamp: now, le: '50' },
+          { value: 15, timestamp: now, le: '100' },
+          { value: 5, timestamp: now, le: '200' },
+          { value: 0, timestamp: now, le: '300' },
+          { value: 0, timestamp: now, le: '500' },
+          { value: 0, timestamp: now, le: '1000' },
+          { value: 0, timestamp: now, le: '+Inf' },
+        ],
```

Comment:
this is awesome. I wonder if we can also add points in the next window so we know it works with multiple timestamp buckets